### DISCUSSION TITLE
Bootloader for Matek H743 flight controller?

### DISCUSSION THREAD

**Discussion by johnnysolar** [qqqlab/madflight#7] [Q&A] [answered]
Hello!  This is a great project - thanks!

I have a Matek H743 flight controller that I've been trying to get this running with, but I'm running into an issue that isn't quite in the scope of your project, but I thought you might be able to help with.
I can compile and run code - either this project, or other very simple projects like just blinking the led.
When I download the binary to the flight controller in DFU mode, it completes and the project starts running (LED blinks as it should), but the computer doesn't detect it as a USB device anymore.  If I unplug and restart without being in DFU mode, it still isn't recognized as a usb device and even the simple blink program never starts running.
This hardware was previously running Betaflight and ardupilot (both of which I've tested and I can still load back on and run).

Is there a bootloader I should be burning?  

Any other tips?

Thanks!

**Comment by qqqlab** [comment, marked as answer]
Hi @johnnysolar, thanks for trying out madflight!

No, this project does not have a bootloader, so for each upload you need to keep the BOOT button pressed and then connect the USB cable.

I did a quick test on a Matek WLITE H743 which had Ardupilot on it, and it worked fine. After programming the first time, I had to reconnect the Arduino IDE to a different serial port in order to get 'Serial' output over USB.

These are the settings I used:

![image](https://github.com/qqqlab/madflight/assets/46283638/e1a72acb-8c34-44d0-93b5-1d17ef3b11ce)

**Reply by johnnysolar** [reply]
Thanks so much for the quick response!

Turns out I had two issues that had nothing to do with madflight.  First was a driver issue with the usb serial port, and second - i think - was that I was using "Generic H743XIHx."  There's a little blob of glue that partly obscures the part #, and it looks like I chose wrong.

Working now, though!

Really nicely done project, btw.  I did something similar based on dRehmFlight a number of years ago, but you've done a much cleaner job!
